apps/web/src/app/organizations/manage/people.component.ts

```diff
@@ -240,36 +241,29 @@ export class PeopleComponent
   }
 
   async edit(user: OrganizationUserUserDetailsResponse) {
-    const [modal] = await this.modalService.openViewRef(
-      UserAddEditComponent,
-      this.addEditModalRef,
-      (comp) => {
-        comp.name = this.userNamePipe.transform(user);
-        comp.organizationId = this.organizationId;
-        comp.organizationUserId = user != null ? user.id : null;
-        comp.usesKeyConnector = user?.usesKeyConnector;
-        // eslint-disable-next-line rxjs-angular/prefer-takeuntil
-        comp.onSavedUser.subscribe(() => {
-          modal.close();
-          this.load();
-        });
-        // eslint-disable-next-line rxjs-angular/prefer-takeuntil
-        comp.onDeletedUser.subscribe(() => {
-          modal.close();
-          this.removeUser(user);
-        });
-        // eslint-disable-next-line rxjs-angular/prefer-takeuntil
-        comp.onRevokedUser.subscribe(() => {
-          modal.close();
-          this.load();
-        });
-        // eslint-disable-next-line rxjs-angular/prefer-takeuntil
-        comp.onRestoredUser.subscribe(() => {
-          modal.close();
-          this.load();
-        });
-      }
-    );
+    const dialog = openUserAddEditDialog(this.dialogService, {
+      data: {
+        name: this.userNamePipe.transform(user),
+        organizationId: this.organizationId,
+        organizationUserId: user != null ? user.id : null,
+        usesKeyConnector: user?.usesKeyConnector,
+      },
+    });
+
+    const result = await lastValueFrom(dialog.closed);
+    switch (result) {
+      case MemberDialogResult.Saved:
+        this.load();
+        break;
+      case MemberDialogResult.Deleted:
+        this.removeUser(user);
+        break;
+      case MemberDialogResult.Revoked:
+        this.load();
+        break;
+      case MemberDialogResult.Restored:
+        this.load();
+    }
```

Comment:
suggestion: We can reduce some of the repeated `this.load()` calls by combining some of these cases.
```suggestion
    switch (result) {
      case MemberDialogResult.Deleted:
        this.removeUser(user);
        break;
      case MemberDialogResult.Saved:
      case MemberDialogResult.Revoked:
      case MemberDialogResult.Restored:
        this.load();
        break;
    }
```

Reply:
Huh, I thought I tried this and got hit with `eslint(no-fallthrough)` but apparently your suggestion works fine 🤷 
I must've missed a `break;` statement or something, thanks for pointing this out!

Reply:
Done!